### PR TITLE
#973 refactoring setter wrapper and update wrapper

### PR DESCRIPTION
**Pull request by sjaakd**

### CODE REVIEW

**Comment by sjaakd** [comment]
There are strange null checks in the calling the update methods when using expressions / constants. Need to check that.

File: processor/src/main/resources/org/mapstruct/ap/internal/model/macro/CommonMacros.ftl [hunk 1]
```diff
@@ -93,15 +95,45 @@ Performs a default assignment with a default value.
...
 -->
-<#macro handleWriteAccesing>
-    <#t><#if fieldAssignment><#else>()</#if>
-</#macro>
\ No newline at end of file
+<#macro handleWriteAccesing><#if fieldAssignment><#else>()</#if></#macro>
+<#--
+  macro: initTargetObject
+
+  purpose: To factoryze or construct a new target object
```
Comment (gunnarmorling) [inline processor/src/main/resources/org/mapstruct/ap/internal/model/macro/CommonMacros.ftl]: factoryze=factorize

File: processor/src/main/java/org/mapstruct/ap/internal/model/assignment/SetterWrapper.java [hunk 2]
```diff
@@ -22,19 +22,41 @@
...
 public class SetterWrapper extends AssignmentWrapper {
 
     private final List<Type> thrownTypesToExclude;
+    private final boolean includeNullCheck;
+
+    public SetterWrapper(Assignment rhs, List<Type> thrownTypesToExclude, NullValueCheckStrategyPrism nvms,
+        boolean fieldAssignment, Type targetType) {
+        super( rhs, fieldAssignment );
+        this.thrownTypesToExclude = thrownTypesToExclude;
+        this.includeNullCheck = !rhs.getSourceReference().equals( rhs.getSourceParameterName() )
```
Comment (filiphr) [inline processor/src/main/java/org/mapstruct/ap/internal/model/assignment/SetterWrapper.java]: I think that the checks should go in some method and maybe somehow add the list items that are there in the class Javadoc.

File: processor/src/main/java/org/mapstruct/ap/internal/model/assignment/GetterWrapperForCollectionsAndMaps.java [hunk 3]
```diff
@@ -41,20 +40,17 @@
     /**
      * @param decoratedAssignment
      * @param thrownTypesToExclude
-     * @param existingVariableNames
      * @param targetType
      * @param fieldAssignment
      */
     public GetterWrapperForCollectionsAndMaps(Assignment decoratedAssignment,
```
Comment (filiphr) [inline processor/src/main/java/org/mapstruct/ap/internal/model/assignment/GetterWrapperForCollectionsAndMaps.java]: I noticed that in the `AdderWrapper` and `SetterWrapper` you renamed the `decoratedAssignment` to `rhs`. Should we do the renaming everywhere (the other `AssignmentWrapper`(s), and the `AssignmentWrapper` itself), or it does not make sense?

File: processor/src/main/resources/org/mapstruct/ap/internal/model/macro/CommonMacros.ftl (hunk 1 above)
Comment (sjaakd) [inline processor/src/main/resources/org/mapstruct/ap/internal/model/macro/CommonMacros.ftl]: will fix.

File: processor/src/main/java/org/mapstruct/ap/internal/model/assignment/SetterWrapper.java (hunk 2 above)
Comment (sjaakd) [inline processor/src/main/java/org/mapstruct/ap/internal/model/assignment/SetterWrapper.java]: will fix.

File: processor/src/main/java/org/mapstruct/ap/internal/model/assignment/GetterWrapperForCollectionsAndMaps.java (hunk 3 above)
Comment (filiphr) [inline processor/src/main/java/org/mapstruct/ap/internal/model/assignment/GetterWrapperForCollectionsAndMaps.java]: OK no problem. Completely agree with your reasoning.

**Comment by sjaakd** [comment]
oops

**Pull request merged**